scipy/sparse/tests/test_base.py

```diff
@@ -68,7 +69,7 @@ def _underlying_arrays(x):
 
         arrays = []
         for a in x.__dict__.values():
-            if isinstance(a, (np.ndarray, np.generic)):
+            if isinstance(a, np.ndarray | np.generic):
```

Comment:
This was "caught" by the linter. Apparently, the type annotations movement has asked Python to add the `__or__` operator to types. That changes the possible syntax/idioms to use for `isinstance` when checking more than one type. It is rather annoying to have the linter switch to checking this without preemptively going through the codebase and changing the existing violations of the codes. But it happened.  Many of our PRs are failing the linting (and thus canceling the other tests) due to `isinstance` that are near (but not related to) the code we are changing. So, I've made the requested changes. But I have to say, I find it harder to read the `|` syntax than the tuple syntax. We removed this check from the NetworkX linting.

So, The `__or__` operation is many versions of python old for classes/types, but newly added to the lint checker.

Reply:
> It is rather annoying to have the linter switch to checking this without preemptively going through the codebase and changing the existing violations of the codes. 

Indeed, I just ran into this in https://github.com/scipy/scipy/pull/21301

It would be nice to fix all the violations *when the lint check is enabled*, instead of leaving lint landmines that no one hits until they happen to touch a file for unrelated reasons.

Reply:
> It would be nice to fix all the violations when the lint check is enabled, instead of leaving lint landmines that no one hits until they happen to touch a file for unrelated reasons.

Strongly agree. Just ran into this elsewhere, too.

Reply:
Another annoying thing: using `|` is the slowest option:

```
In [16]: %timeit isinstance(float, (list, tuple))
56.1 ns ± 0.189 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [17]: %timeit isinstance(float, list | tuple)
102 ns ± 0.272 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [18]: %timeit isinstance(float, list) or isinstance(float, tuple)
51.6 ns ± 0.282 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```